src/utils/IBCERC20.sol

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.25;
+
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { IICS20Transfer } from "../interfaces/IICS20Transfer.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+contract IBCERC20 is ERC20, Ownable {
+    // TODO: Figure out naming and symbol for IBC denoms
```

Comment:
You mean the ibc/{hash(trace+base_denom)} think we use for the denom? We can do that here too, for sure. I just wonder if it is necessary or provide a lot of value, compared to actually having the trace available directly in the name (less hashing to do, less lookups for both us and others).

Reply:
We must do it here since spec says **MUST**.

Reply:
The spec only says `It must encode the trace and base denomination to ensure that tokens coming over different paths are not treated as fungible.`, which the full denom does, but it takes more space than a hash.

Reply:
I see, but still, these minted tokens only have the name and symbol "IBC". We need to encode the info in the name and symbol right?

Reply:
I'd actually propose we use ibc-go's implementation regardless as it has been battle tested for so long. I don't want us to do our own implementation and mess something up with some type of collision or a denom that contains slashes or something like that.

Reply:
Yes, we do need to add the info there. We could add the full denom as the name, as it might be harder to look up the full trace (given that the erc20 itself doesn't contain any link back to the ICS20Transfer contract where that information would most likely be). We could also use a hash as the name and have a separate denom or even trace with hops field like in the new Denom struct.

Reply:
The symbol should be fine, will do what the ibc-go implementation does

Reply:
"mess something up with some type of collision or a denom that contains slashes or something like that." 
Note that in the EVM world we have to consider that we may event different hash collision problems to consider. Another note for future thoughts and audits.